### PR TITLE
Add Trim to String plugin

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes

This PR adds the Trim action to the String plugin

## Tests: 
```
rmt-mbp-5357:string jmcadams$ ./run.sh -R tests/trim.json -j
Running: cat tests/trim.json | docker run --rm   -i rapid7/string:1.1.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
rapid7/String Operations:1.1.0. Step name: trim
rapid7/String Operations:1.1.0. Step name: trim

{
  "trimmed": "this is a string"
}
```

## Validation: 
```
rmt-mbp-5357:string jmcadams$ make validate
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Colors.mk ../tools/Makefiles/Helpers.mk
--
[*] Running validators
WARNING: You are using pip version 19.2.1, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator SpecPropertiesValidator
[*] Executing validator FilesValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator HelpValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator DockerValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator ExceptionValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 10326.884ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	running on Python 2.7.15
Run started:2019-11-11 22:19:40.790975

Test results:
	No issues identified.

Code scanned:
	Total lines of code: 508
	Total lines skipped (#nosec): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
	Total issues (by confidence):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
Files skipped (0):
[SUCCESS] Passes bandit security checks
```

## Workflow: 
image

### Artifacts: 
image

### Input: 
image

### Output: 
image

### CODE REVIEW

**Comment by jmcadams-r7** [comment]
This addresses DF-3564

File: string/komand_string/actions/trim/action.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='trim',
+                description=Component.DESCRIPTION,
+                input=TrimInput(),
+                output=TrimOutput())
+
+    def run(self, params={}):
+        string = params.get(Input.STRING)
+        if not string:
```
Comment (mrinehart-r7) [inline string/komand_string/actions/trim/action.py]: No need to check for this, the input is required

File: string/plugin.spec.yaml [hunk 2]
```diff
@@ -127,3 +127,18 @@ actions:
         title: Lowercase
         description: Lowercase string
         required: false
+  trim:
+    title: Trim
+    description: Trim a string of leading or trailing whitespace
```
Comment (mrinehart-r7) [inline string/plugin.spec.yaml]: Leading trailing, there's no option for either or

File: string/plugin.spec.yaml [hunk 3]
```diff
@@ -127,3 +127,18 @@ actions:
...
+      string:
+        title: String Input
+        description: String to trim
+        type: string
+        required: true
+    output:
+      trimmed:
+        type: string
+        title: Trimmed
+        description: Trimmed String
```
Comment (mrinehart-r7) [inline string/plugin.spec.yaml]: Trimmed string

File: string/plugin.spec.yaml [hunk 4]
```diff
@@ -127,3 +127,18 @@ actions:
...
+        title: String Input
+        description: String to trim
+        type: string
+        required: true
+    output:
+      trimmed:
+        type: string
+        title: Trimmed
+        description: Trimmed string
+        required: false
```
Comment (mrinehart-r7) [inline string/plugin.spec.yaml]: Required should be true

File: string/komand_string/actions/trim/action.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import komand
+from .schema import TrimInput, TrimOutput, Input, Output, Component
+# Custom imports below
+from komand.exceptions import PluginException
```
Comment (mrinehart-r7) [inline string/komand_string/actions/trim/action.py]: Unnecessary import
